cirq/sim/wave_function.py

```diff
@@ -282,3 +286,40 @@ def _validate_indices(num_qubits: int, indices: List[int]) -> None:
     if any(index >= num_qubits for index in indices):
         raise IndexError('Out of range indices, must be less than number of '
                          'qubits but was {}'.format(indices))
+
+
+def sample_terminal_measurements(
+        circuit: circuits.Circuit,
+        last_step_result: 'simulator.StepResult',
+        repetitions: int) -> Dict[str, List]:
+    """Sample from measurements in the given circuit.
+
+    This should only be called if the circuit has only terminal measurements.
```

Comment:
Explain why

Reply:
Moved this into step result where it has a different meaning which should make it clear